File: InventoryDemo.DTOs/Responses/PropertyListResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace InventoryDemo.DTOs.Responses
{
    public class PropertyListResponse
    {
        public int ID { get; set; }
        public string Name { get; set; }
        public string? ImageURL { get; set; }
        public int Quantity { get; set; }
        public string? FullDetail { get; set; }
        public string? ShortDescription { get; set; }
        public int CategoryID { get; set; }
    }
}

File: lib/Models/Property/PropertyListResponse.dart
class PropertyListResponse {
  final int id;
  final String name;
  final String? imageURL;
  final int quantity;
  final String? fullDetail;
  final String? shortDescription;
  final int categoryID;

  const PropertyListResponse(
      {required this.id,
      required this.name,
      required this.imageURL,
      required this.quantity,
      required this.fullDetail,
      required this.shortDescription,
      required this.categoryID});

  factory PropertyListResponse.fromJson(Map<String, dynamic> jsonData) {
    return PropertyListResponse(
        id: jsonData["id"],
        name: jsonData["name"],
        imageURL: jsonData["imageURL"],
        quantity: jsonData["quantity"],
        fullDetail: jsonData["fullDetail"],
        shortDescription: jsonData["shortDescription"],
        categoryID: jsonData["categoryID"]);
  }
}
